blog/02-18-2023-Build-an-NFT-Full-Stack-Monster-War-Game-on-The-Celo-Blockchain/Build-an-NFT-Full-Stack-Monster-War-Game-on-The-Celo-Blockchain.md

```diff
@@ -0,0 +1,998 @@
+            nfts[_index] = nfts[allNFTs - 1];
+            delete nfts[allNFTs - 1];
+            allNFTs--; 
+            _transfer(address(this), msg.sender, nfts[_index].tokenId);
+	 }
+```
+
+The `Remove()` function removes an NFT from the game. The function takes an `index` as a parameter which is used to specify the NFT to be removed. The function requires that the sender of the transaction is the owner of the NFT. If this requirement is met, the function transfers the NFT from the contract to the owner and deletes the NFT from the internal nfts mapping. 
+
+```solidity
+ function getNFTlength() public view returns (uint256) {
+        return allNFTs;
+    }
+```
+
+The `GetNFTlength()` function returns the length of the NFTs in the list. The function returns the value of the `allNFTs` variable which is the length of all monster NFTs in the list. 
+
+The rest of the functions are overrides that are required by solidity.
+
+That’s it For the smart contract. Next, we’ll be looking at the front end.
+
+## Front end
+
+### Stack
+
+We’ll use the following stack for this section.
+
+Hardhat
+useContractKit
+
+### Setup
+
+Clone the full project from [this Repository](https://github.com/4undRaiser/celo-monster-nft-game) to follow up with this section.
+
+Here is an example of the env file
+
+```.env
+MNEMONIC=""
+REACT_APP_STORAGE_API_KEY=""
+```
+
+### Deployment
+
+We’ll use hardhat to deploy our smart-contracts to the celo blockchain.
+
+```javascript
+require("@nomiclabs/hardhat-waffle");
+require("dotenv").config({ path: ".env" });
+
+// This is a sample Hardhat task. To learn how to create your own go to
+// https://hardhat.org/guides/create-task.html
+task("accounts", "Prints the list of accounts", async (taskArgs, hre) => {
+  const accounts = await hre.ethers.getSigners();
+
+  for (const account of accounts) {
+    console.log(account.address);
+  }
+});
+
+const getEnv = (variable, optional = false) => {
+  if (!process.env[variable]) {
+    if (optional) {
+      console.warn(
+        `[@env]: Environmental variable for ${variable} is not supplied.`
+      );
+    } else {
+      throw new Error(
+        `You must create an environment variable for ${variable}`
+      );
+    }
+  }
+
+  return process.env[variable]?.replace(/\\n/gm, "\n");
+};
+
+// Your mnemomic key
+const MNEMONIC = getEnv("MNEMONIC");
+
+// You need to export an object to set up your config
+// Go to https://hardhat.org/config/ to learn more
+
+/**
+ * @type import('hardhat/config').HardhatUserConfig
+ */
+module.exports = {
+  solidity: "0.8.9",
+  networks: {
+    alfajores: {
+      url: "https://alfajores-forno.celo-testnet.org",
+      accounts: {
+        mnemonic: process.env.MNEMONIC,
+        path: "m/44'/52752'/0'/0",
+      },
+      chainId: 44787,
+    },
+  },
+};
+```
+
+Configure your `hardhat.config` file to look like this to enable hardhat to deploy the smart contracts to the celo blockchain.
```

Comment:
If this is the description of the location of the code above, It should come first before the code snippet